pkg/pillar/containerd/oci.go

```diff
@@ -535,3 +531,24 @@ func (s *ociSpec) UpdateEnvVar(envVars map[string]string) {
 		}
 	}
 }
+
+// UpdateShimVMDevices grants full access to file devices for shim VM OCI specs
+func (s *ociSpec) UpdateShimVMDevices() {
+	if len(s.Get().Process.Args) == 0 {
+		return
+	}
+
+	ctrCmd := s.Get().Process.Args[0]
+
+	// default OCI specs have all devices being denied by default, check for a
+	// shim VM loader (qemu/xen) and enable full access to file devices on
+	// these cases. Later on we may need to get more fine-grained.
+	if strings.Contains(ctrCmd, "qemu-system") ||
```

Comment:
Wouldn't it be nicer if this check happens not inside the ` UpdateShimVMDevices()` function but before it call?
Something like:
```go
      if s.isShimVM()
	// Set file device access for ShimVM based containers
	s.UpdateShimVMDevices()
```
and the isShimVM(), which contains the logic you provide here...

Reply:
It can be done, but see my [comment](https://github.com/lf-edge/eve/pull/4211/files#r1744156091) for Christoph on the rationale about the UpdateShimVMDevices() function.